### PR TITLE
[node-manager] Prevent node deletion by autoscaler

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

Prevent node (with CloudPermanent or Static type) deletion by autoscaler.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: fix
summary: Prevent node (with `CloudPermanent` or `Static` type) deletion by autoscaler.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

**Review by name212 — APPROVED** [review]

File: modules/040-node-manager/hooks/handle_node_templates.go [hunk 1]
```diff
@@ -193,6 +193,11 @@ func nodeTemplatesHandler(input *go_hook.HookInput) error {
 				}
 			}
 
+			// Prevent node deletion by autoscaler
+			if nodeObj.Labels["node.deckhouse.io/type"] == "CloudPermanent" || nodeObj.Labels["node.deckhouse.io/type"] == "Static" {
```
Comment (name212) [inline modules/040-node-manager/hooks/handle_node_templates.go:197]: CloudStatic also

**Review by name212 — CHANGES_REQUESTED** [review]
one thing

File: modules/040-node-manager/hooks/handle_node_templates.go (hunk 1 above)
Comment (RomanenkoDenys) [inline modules/040-node-manager/hooks/handle_node_templates.go:197]:
```suggestion
			if set.New("CloudPermanent", "CloudStatic", "Static").Has(nodeObj.Labels["node.deckhouse.io/type"]) {
```

Little deckhouse magic

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
one thing

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

File: modules/040-node-manager/hooks/handle_node_templates.go (hunk 1 above)
Comment (vitaliy-sn) [reply modules/040-node-manager/hooks/handle_node_templates.go]: done

Comment (vitaliy-sn) [reply modules/040-node-manager/hooks/handle_node_templates.go]: done

**Review by name212 — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**